packages/expo/src/winter/TextDecoder.ts

```diff
@@ -0,0 +1,429 @@
+        this.utf8BytesNeeded = 1;
+
+        // 2. Set UTF-8 code point to byte & 0x1F.
+        this.utf8CodePoint = bite & 0x1f;
+      }
+
+      // 0xE0 to 0xEF
+      else if (inRange(bite, 0xe0, 0xef)) {
+        // 1. If byte is 0xE0, set utf-8 lower boundary to 0xA0.
+        if (bite === 0xe0) this.utf8LowerBoundary = 0xa0;
+        // 2. If byte is 0xED, set utf-8 upper boundary to 0x9F.
+        if (bite === 0xed) this.utf8UpperBoundary = 0x9f;
+        // 3. Set utf-8 bytes needed to 2.
+        this.utf8BytesNeeded = 2;
+        // 4. Set UTF-8 code point to byte & 0xF.
+        this.utf8CodePoint = bite & 0xf;
+      }
+
+      // 0xF0 to 0xF4
+      else if (inRange(bite, 0xf0, 0xf4)) {
+        // 1. If byte is 0xF0, set utf-8 lower boundary to 0x90.
+        if (bite === 0xf0) this.utf8LowerBoundary = 0x90;
+        // 2. If byte is 0xF4, set utf-8 upper boundary to 0x8F.
+        if (bite === 0xf4) this.utf8UpperBoundary = 0x8f;
+        // 3. Set utf-8 bytes needed to 3.
+        this.utf8BytesNeeded = 3;
+        // 4. Set UTF-8 code point to byte & 0x7.
+        this.utf8CodePoint = bite & 0x7;
+      }
+
+      // Otherwise
+      else {
+        // Return error.
+        return decoderError(this.options.fatal);
+      }
+
+      // Return continue.
+      return null;
+    }
+
+    // 4. If byte is not in the range utf-8 lower boundary to utf-8
+    // upper boundary, inclusive, run these substeps:
+    if (!inRange(bite, this.utf8LowerBoundary, this.utf8UpperBoundary)) {
+      // 1. Set utf-8 code point, utf-8 bytes needed, and utf-8
+      // bytes seen to 0, set utf-8 lower boundary to 0x80, and set
+      // utf-8 upper boundary to 0xBF.
+      this.utf8CodePoint = 0;
+      this.utf8BytesNeeded = 0;
+      this.utf8BytesSeen = 0;
+      this.utf8LowerBoundary = 0x80;
+      this.utf8UpperBoundary = 0xbf;
+
+      // 2. Prepend byte to stream.
+      stream.prepend(bite);
+
+      // 3. Return error.
+      return decoderError(this.options.fatal);
+    }
+
+    // 5. Set utf-8 lower boundary to 0x80 and utf-8 upper boundary
+    // to 0xBF.
+    this.utf8LowerBoundary = 0x80;
+    this.utf8UpperBoundary = 0xbf;
+
+    // 6. Set UTF-8 code point to (UTF-8 code point << 6) | (byte &
+    // 0x3F)
+    this.utf8CodePoint = (this.utf8CodePoint << 6) | (bite & 0x3f);
+
+    // 7. Increase utf-8 bytes seen by one.
+    this.utf8BytesSeen += 1;
+
+    // 8. If utf-8 bytes seen is not equal to utf-8 bytes needed,
+    // continue.
+    if (this.utf8BytesSeen !== this.utf8BytesNeeded) return null;
+
+    // 9. Let code point be utf-8 code point.
+    const code_point = this.utf8CodePoint;
+
+    // 10. Set utf-8 code point, utf-8 bytes needed, and utf-8 bytes
+    // seen to 0.
+    this.utf8CodePoint = 0;
+    this.utf8BytesNeeded = 0;
+    this.utf8BytesSeen = 0;
+
+    // 11. Return a code point whose value is code point.
+    return code_point;
+  }
+}
+
+// 8.1 Interface TextDecoder
+
```

Comment:
```suggestion
// @docsMissing
```

Let's annotate missing description of exported class, or fill it up.